### PR TITLE
Add ServiceAccount, Role, RoleBinding, and PSP for using EmptyDir

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/7349

**This PR does not contain any changes regarding the Giant Swarm API specification. The focus is on the technicalities of publishing the documentation.**

This PR should enable the app to use an EmptyDir volume for the nginx cache. It's derived from the working docs deployment in gollum/35ac3.

### CODE REVIEW

**Comment by marians** [comment]
Would be kind if you could just check for any typos

File: helm/api-spec-app/templates/psp-rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+apiVersion: extensions/v1beta1
```
Comment (oponder) [inline helm/api-spec-app/templates/psp-rbac.yaml]:
```suggestion
apiVersion: policy/v1beta1
```

File: helm/api-spec-app/templates/psp-rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+kind: PodSecurityPolicy
+metadata:
+    name: {{ .Values.name }}
+spec:
+    allowPrivilegeEscalation: false
+    fsGroup:
```
Comment (oponder) [inline helm/api-spec-app/templates/psp-rbac.yaml]: I feel like theres some indentation wrong here.

Comment (oponder) [reply helm/api-spec-app/templates/psp-rbac.yaml]: and in other places (like seLinux for example)

File: helm/api-spec-app/templates/psp-rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    rule: RunAsAny
+    supplementalGroups:
+    ranges:
+    - max: 65535
+        min: 1
+    rule: MustRunAs
+    volumes:
+    - secret
+    - configMap
+    - emptyDir
```
Comment (calvix) [inline helm/api-spec-app/templates/psp-rbac.yaml]: the indentation in psp is broken, check this for reference https://github.com/giantswarm/happa/blob/master/helm/happa-chart/templates/psp.yaml

Comment (marians) [reply helm/api-spec-app/templates/psp-rbac.yaml]: Fixed

File: helm/api-spec-app/templates/psp-rbac.yaml (hunk 1 above)
Comment (marians) [reply helm/api-spec-app/templates/psp-rbac.yaml]: Done

File: helm/api-spec-app/templates/psp-rbac.yaml (hunk 2 above)
Comment (marians) [reply helm/api-spec-app/templates/psp-rbac.yaml]: Done

**Comment by marians** [comment]
@oponder Can you re-think your review, please? 🌻

**Pull request merged**